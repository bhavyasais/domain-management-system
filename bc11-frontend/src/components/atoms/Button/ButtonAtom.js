import React from "react";
import Button from "@material-ui/core/Button";
import { makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import clsx from "clsx";
const useStyles = makeStyles({
  buttonLabel: {
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
});

export default function ButtonAtom({
  value,
  variant,
  disabled,
  color,
  onClick,
  textColor,
  rootStyle,
  labelStyle,
  textVariant,
  maxWidth,
  ...restProps
}) {
  const classes = useStyles();
  return (
    <div>
      <Button
        variant={variant}
        disabled={disabled}
        style={{ backgroundColor: color, maxWidth: maxWidth }}
        onClick={onClick}
        classes={{
          root: rootStyle,
          label: clsx(labelStyle, classes.buttonLabel),
        }}
        {...restProps}
      >
        <Typography variant={textVariant} className={classes.buttonLabel}>
          {value}
        </Typography>
      </Button>
    </div>
  );
}
Button.propTypes = {
  value: PropTypes.string,
  color: PropTypes.string,
  variant: PropTypes.string,
};
Button.defaultProps = {
  variant: "contained",
};
