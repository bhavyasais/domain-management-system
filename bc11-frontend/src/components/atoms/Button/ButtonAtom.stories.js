import React from "react";
import { storiesOf } from "@storybook/react";
import ButtonAtom from "./ButtonAtom";
import { text ,boolean} from "@storybook/addon-knobs";

storiesOf("Atoms/Button", module).add("Contained Button", () => (
  <ButtonAtom
    variant={text("Button Variant","contained")}
    value={text("Text", "Add")}
    // color={color("Color","#1d4cd7")}
    //textVariant={text("Typography Variant","h4")}
    onClick={() => console.log("Button clicked")}
    disabled={boolean('Disabled', false)}
    maxWidth="100px"
    disableRipple
  />));
