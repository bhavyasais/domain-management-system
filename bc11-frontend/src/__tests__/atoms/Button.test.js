import Button from "../../components/atoms/Button/ButtonAtom";
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";

describe("Button", () => {
  const handleClick = jest.fn();
  let component;
  beforeEach(() => {
    act(() => {
      component = mount(<Button variant="contained" value="Add" onClick={handleClick} />);
    });
  });

  it("should render correctly", () => {
    let wrapper;
    wrapper = renderer.create(<Button variant="contained" value="Add" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have the text Add", () => {
      expect(component.find(Button).text()).toEqual("Add");
  });

  it("should have the variant contained", () => {
    expect(component.find(Button).at(0).props().variant).toBe("contained");
  });

  it("should call the onClick function when clicked", () => {
    component.find(Button).at(0).props().onClick();
    expect(handleClick).toHaveBeenCalledTimes(1);
  })

});
