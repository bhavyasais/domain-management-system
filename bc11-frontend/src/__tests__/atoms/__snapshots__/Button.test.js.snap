// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
        "maxWidth": undefined,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label makeStyles-buttonLabel-1"
    >
      <p
        className="MuiTypography-root makeStyles-buttonLabel-1 MuiTypography-body1"
      >
        Add
      </p>
    </span>
  </button>
</div>
`;
